### DISCUSSION TITLE
ReactApp web config for BlueRetro

### DISCUSSION THREAD

**Discussion by delislej** [darthcloud/BlueRetro#297] [Show and tell]
Hi there, i decided to make a nice web interface for the blueretro project! please come check it out and let me know if you run into any problems! oh and it works nicely with chrome on android!

https://blueretro.delisle.io/

like the regular blueretro.io you will need to use chrome or edge, and allow bluetooth for web browsers!

features include:

-Advanced settings lets you configure controller settings such as rumble and special controller types. It also allows for advance settings such as deep sleep and Factory Reset

-N64 controller pak manager lets you read/write/format your N64 controller paks.

-Presets lets you flash a controller preset.

-Presets maker will let you make and use custom presets right from your phone. these presets can be downloaded and shared with others!

the source for this project can be found at https://github.com/delislej/BlueRetro-ReactApp

**Comment by delislej** [comment]
please feel free to break it and let me know where problems are!

**Reply by darthcloud** [reply]
Awesome work congrats!

**Comment by bbsan2k** [comment]
Looks very nice!
Just for curiosity: Are you planning to also add an option to flash the ESP through UART directly using esp-web-tools?

**Reply by delislej** [reply]
I'm very much just a web dev guy. Most of the Bluetooth work with the ESP32 is black box work where I know the proper output, but I'm not really sure how/why it works 😂. If Darth adds it to his website I can probably adapt it to react.

What I might do going forward is making my own n64 memory pak editor based on brycs mempak editor.

Longer term I might adapt this into a react-native app for Android/ios.

**Reply by darthcloud** [reply]
Using esp-web-tools for BlueRetro is not as straightforward as it might look.

Nothing big but still require to invest some time into it. (Which I likely won't TBH)

* Need to check if one manifest can support multiple build variation. It does support multiple chip but it's not what's needed for blueretro. You could use multiple different manifest I guess.
* BlueRetro release are zip file of all the build together, I like that a lot as it provide data integrity without external checksum file.
  I'm pretty much against putting binary file in a git repo.
  Syncing the release on a file server sound annoying.
  So a requirement for the web tool is to downloads the zip and extract the file it's need in a js byte array

While this sound nice in practice how hard is it to use the windows tools anyway?
Its a one time thing to do for DIY user after that you use the OTA update.
For people buying dongle prebuilt it's not even an issue.

**Comment by darthcloud** [comment]
@delislej I merged the utils module part in https://darthcloud.github.io/BlueRetroWebCfg_Beta/index.html

Thanks a lot for this, it save me a lot of work and now the sources are much more organized!

I'll leave in beta repo for a while and eventually merge it in the main one.
